src/runtime/storageNG/store.ts

```diff
@@ -79,6 +91,35 @@ export class DirectStore<T extends CRDTTypeRecord> extends ActiveStore<T> {
     super(storageKey, exists, type, mode, modelConstructor);
   }
 
+  async idle() {
+    if (this.pendingException) {
+      return Promise.reject(this.pendingException);
+    }
+    if (this.state === DirectStoreState.Idle) {
+      return Promise.resolve();
+    }
+    return new Promise<void>((resolve, reject) => {
+      this.pendingResolves.push(resolve);
+      this.pendingRejects.push(reject);
+    });
+  }
+
+  private setState(state: DirectStoreState) {
+    this.state = state;
+    if (state === DirectStoreState.Idle) {
+      this.notifyIdle();
```

Comment:
it's OK to notify when going idle -> idle because that won't actually have any effect (anything registered while idle immediately notifies, anything registered will have already been triggered by the first transition to idle).